src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -62,34 +70,58 @@
 @Component
 public class NotificationService implements Job {
```

Comment:
In general, there has been some refactoring to lift common functionality out of the PlacementService and the ProgrammeMembershipService and hold them here.